File: Frontend-React/src/types/Notification.ts
import { NotificationType } from './NotificationType';

export interface Notification {
    id: number;
    userId: number;
    message: string;
    type: NotificationType;
    isRead: boolean;
    createdAt: string;
    relatedEntityId?: number;
    relatedEntityType?: string;
} 

File: Backend-dotnet/Models/Notification.cs
using System;

namespace NotepadPlusApi.Models;

public enum NotificationType
{
    NewPublicNote,
    NoteShared
}

public class Notification
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    public string Message { get; set; } = string.Empty;
    public NotificationType Type { get; set; }
    public int? NoteId { get; set; }
    public Note? Note { get; set; }
    public DateTime CreatedAt { get; set; }
    public bool IsRead { get; set; }
} 